### DISCUSSION TITLE
Save language selction local storage or cookie

### DISCUSSION THREAD

**Discussion by MuhammadSaim** [vinissimus/next-translate#1201] [Q&A] [answered]
How can I save the selected language in local storage or cookie when a user comes back and according to the cookie or value in storage selects his/her selected language?

**Comment by aralroca** [comment, marked as answer]
I hope this documentation helps: https://github.com/aralroca/next-translate?tab=readme-ov-file#11-how-to-save-the-user-defined-language:

```ts
import { useRouter } from 'next/router'

// ...

function usePersistLocaleCookie() {
    const { locale, defaultLocale } = useRouter()

    useEffect(persistLocaleCookie, [locale, defaultLocale])
    function persistLocaleCookie() {
      if(locale !== defaultLocale) {
         const date = new Date()
         const expireMs = 100 * 24 * 60 * 60 * 1000 // 100 days
         date.setTime(date.getTime() + expireMs)
         document.cookie = `NEXT_LOCALE=${locale};expires=${date.toUTCString()};path=/`
      }
    }
}
```

The cookie name should be `NEXT_LOCALE`, this is part of the core of Next.js: https://nextjs.org/docs/pages/building-your-application/routing/internationalization#leveraging-the-next_locale-cookie

**Reply by MuhammadSaim** [reply]
Thank @aralroca for your reply. I'm using this but on refresh package is not reading the locale from the cookies 
![image](https://github.com/aralroca/next-translate/assets/19898499/9aecc01f-2e01-4d33-aa71-f132762522c6)
